build/dsc.rake

```diff
@@ -73,25 +77,52 @@ eod
       puts "Cleaning out black-listed DSC resources: #{blacklist}"
       blacklist.each { |res| FileUtils.rm_rf("#{dsc_resources_path_tmp}/xDSCResources/#{res}") }
 
+      resource_tags = {}
+      resource_tags = YAML::load_file("#{dsc_resources_file}") if File.exist? dsc_resources_file
+
       puts "Getting latest release tags for DSC resources..."
       Dir["#{dsc_resources_path_tmp}/xDSCResources/*"].each do |dsc_resource_path|
         dsc_resource_name = Pathname.new(dsc_resource_path).basename
         FileUtils.cd(dsc_resource_path) do
           # --date-order probably doesn't matter
           tags_raw = %x{ git log --tags --pretty=format:'%D' --simplify-by-decoration --date-order }
-          # if this starts to result in errors, we should explore pushing this out to a file
-          # would prefer another method over scan
+          # If the conversion of string to version starts to result in errors,
+          # we should explore pushing this out out to a method where we can
+          # clean up the tags that may have prerelease versions in them
+          # similar to what was done in the Chocolatey module
           versions = tags_raw.scan(/(\S+)\-PSGallery/).map { | ver | Gem::Version.new(ver[0]) }
           if versions.empty?
             raise "#{dsc_resource_name} does not have any '*-PSGallery' tags. Appears it has not been released yet. Tags found #{tags_raw.to_s}"
           end
 
           latest_version = versions.max.to_s + "-PSGallery"
-          puts "Latest tag for #{dsc_resource_name} is #{latest_version}. Switching to that tag."
-          sh "git checkout #{latest_version}"
+          tracked_version = resource_tags["#{dsc_resource_name}"]
+
+          update_version = update_versions
+          update_version = true if tracked_version.nil?
+
+          if update_version
+            puts "Latest/available tag for #{dsc_resource_name} is #{latest_version}. Switching to that tag."
+            checkout_version = latest_version
+          else
+            puts "Using tag for #{dsc_resource_name} is #{tracked_version}. Switching to that tag."
+            checkout_version = tracked_version
+          end
+
+          # If the checkout_version is not a standard PSGallery tag, a git fetch
+          # is required before a git checkout e.g. for commits or non-default branch names
+          if !(checkout_version =~ /-PSGallery/)
+            puts "#{checkout_version} is not a PSGallery tag. Fetching from git remote"
+            sh "git fetch"
```

Comment:
No because I would then be depending on DSC_REF variable still.  This way you can use any commit or branch in history.

It seemed such a small change, and it was still in the spirit of the previous commits (using a specific point in time of the repo) already so I didn't see it was required.  Can pull it out to a separate one if required.